oam/install_scripts/mcs-workernode.service.in

```
@@ -13,6 +13,7 @@ LimitCORE=@CORE_DUMPS@
 
 Environment="@WORKERNODE_ALLOC_CONFIG@"
 ExecStart=@ENGINE_BINDIR@/workernode DBRM_Worker%i
+ExecStop=/usr/bin/env bash -c "@ENGINE_BINDIR@/mcs-prestop-workernode.sh; /bin/kill -TERM $MAINPID"
```

I have some notes about it:
1. this implementation hides failures in the .sh script, because the last command overwrites the exit status. It's not an issue If it's expected.
2. it’s not idiomatic systemd

My suggestion is:

```
ExecStop=-@ENGINE_BINDIR@/mcs-prestop-workernode.sh
ExecStop=/bin/kill -TERM $MAINPID
```

The unit still won’t be marked as failed even if .sh failed.
But it's more systemd-idiomatic and each command is tracked separately.

---

Thanks, i move whole workernode-prestop implementation into cmapi